### DISCUSSION TITLE
dynamic number of shared values causes an error

### DISCUSSION THREAD

**Discussion by yanush** [software-mansion/react-native-reanimated#1963] [Q&A]
I have a view with dynamic number of elements inside.
For each element I'm initializing its SharedValue (for translations) inside a loop.
Everything works fine as long as the number of items doesn't change.
Whenever I add another item to my array of items, I get the next error:
"Rendered more hooks than during the previous render."
I have tried instead of using one SharedValue per item, to use a SharedValue object with keys for every item, but failed to do so.
Is there any way to work around this problem?

**Comment by yanush** [comment]
Never mind,
I think I've figured out a way around it

**Reply by likern** [reply]
@yanush How you solved this problem?

**Reply by yanush** [reply]
In my case, instead of using useSharedValue in the loop of the container component, the loop now creates my item components, and inside each item's render I use useSharedValue.
Not sure however it would be easy to use that method for all cases, since sometimes you need the shared value in the container component

**Reply by nknaveen007** [reply]
can you please show me the code

**Reply by yanush** [reply]
It will be hard for me to extract it since it's a big project,
but as i have said, -  instead of having the useSharedValue inside the loop, just move it to inner component.
So for example let's say you have a ListComponent, and it renders an array of ItemComponents, just put useSharedValue inside ItemComponent and not in ListComponent

**Reply by joshua7v** [reply]
I think this is a common scenario, but I cannot find any document about this.
Yes, you can useSharedValue inside ChildComponent to make it work.
But it will get really complicated when you want to control child animations(modify shared values) from outside.

**Reply by yanush** [reply]
To make it easier to control animations from the outside, I save references to children's animate functions in parent's data structure.
For example, parent has an array of N children and another array of N animate functions. Every child has its own animate() function and from child's useEffect I call parent's setAnimateFunctionOfChild(key). Now according to the need, I call the matching child's function that is saved in paren't data structure.
Does it make sense?
